shell/browser/api/electron_api_app.cc

```diff
@@ -1132,11 +1132,17 @@ void App::DisableHardwareAcceleration(gin_helper::ErrorThrower thrower) {
         "before app is ready");
     return;
   }
-  if (content::GpuDataManager::Initialized()) {
+
+  if (content::GpuDataManager::Initialized())
     content::GpuDataManager::GetInstance()->DisableHardwareAcceleration();
-  } else {
-    disable_hw_acceleration_ = true;
-  }
+  disable_hw_acceleration_ = true;
+}
+
+bool App::IsHardwareAccelerationEnabled() {
+  if (content::GpuDataManager::Initialized())
+    return content::GpuDataManager::GetInstance()
+        ->HardwareAccelerationEnabled();
+  return !disable_hw_acceleration_;
```

Comment:
I don't think we should use this flag to determine if hardware acceleration is enabled - unless we change the meaning to mean that it is intended to be disabled in the future.

Reply:
Yeah - i debated this for a while. The problem is that before `content::GpuDataManager::Initialized()` is true, the answer to whether hardware acceleration is enabled isn't really true _or_ false - we don't know. Returning `true` here feels incorrect, because if we're at this point in control flow we don't actually know that. So the most accurate result for me would be imo to return what will be true the moment that `OnPreCreateThreads` executes.